src/main/scala/is/hail/expr/ir/MatrixValue.scala

```diff
@@ -197,12 +197,19 @@ case class MatrixValue(
       val localEntryType = typ.entryType
       val localNCols = nCols
 
-      val localColValues = colValues.broadcast.value
+      val localColValues = colValues.broadcast
 
       rvd.boundary.mapPartitions(resultStruct, { (ctx, it) =>
         val rv2b = ctx.rvb
         val rv2 = RegionValue(ctx.region)
 
+        val colRegion = ctx.freshRegion
+        val colRVB = new RegionValueBuilder(colRegion)
+        val colsType = TArray(localColType, required = true)
+        colRVB.start(colsType)
+        colRVB.addAnnotation(colsType, localColValues.value)
+        val colsOffset = colRVB.end()
```

Comment:
Yeah this is fine because you generate data once-per-partition. It's not necessary to every call clear on this region.